lib/methods/upsert.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Updates a model or creates the model if it cannot be found.
+ * @param {Arrow.Model} Model The model class being updated.
+ * @param {String} id ID of the model to update.
+ * @param {Object} doc Model attributes to set.
+ * @param {Function} callback Callback passed an Error object (or null if successful) and the updated or new model.
+ */
+
+var _ = require('lodash')
+
+exports.upsert = function upsert (Model, id, doc, callback) {
+  var self = this
+  var table = this.getTableName(Model)
+  var payload = Model.instance(doc, false).toPayload()
+  var primaryKeyColumn = this.getPrimaryKeyColumn(Model)
+  var columns = this.fetchColumns(table, payload)
+  var placeholders = columns.map(this.returnPlaceholder)
+  var query
+  var updateQuery = ''
+
+  if (!id || !doc) {
+    throw new Error('You must provide a Model id and data Object, that will be persisted')
+  }
+
+  self.findByID(Model, id, function (err, result) {
+    if (err) {
+      callback(err)
+    } else if (result) {
+      for (var index in doc) {
+        updateQuery = updateQuery + index + " = '" + doc[index] + "', "
+      }
+      updateQuery = updateQuery.slice(0, -2)
+      query = 'UPDATE ' + table + ' SET ' + updateQuery + ' WHERE  ' + primaryKeyColumn + ' = ' + id
+
+      self._query(query, callback, function (result) {
```

Comment:
You should use the logic provided in the save method, which prevents from sql injection, i.e. https://github.com/appcelerator/appc.mysql/blob/master/lib/methods/save.js#L15-L22